src/main/java/com/edmunds/rest/databricks/service/ClusterServiceImpl.java

```diff
@@ -125,4 +153,18 @@ public ClusterInfoDTO getInfo(String clusterId) throws IOException, DatabricksRe
     ClusterEventsDTO clusterEvents = this.mapper.readValue(responseBody, ClusterEventsDTO.class);
     return clusterEvents.getEvents();
   }
+
+  @Override
+  public void upsertCluster(UpsertClusterDTO clusterDTO) throws IOException, DatabricksRestException {
+    String clusterId = clusterDTO.getClusterId();
```

Comment:
Could we modify the upsertClusterDTO to take a clusterName (or perhaps we should support both) and then do a lookup here?
Ideally, the cluster configurations are configured by name and not by id so we can use the same name in multiple environments.

Reply:
Done, now we do lookup within upsertCluster method